File: eshoppers/src/main/java/com/shihabmahamud/eshoppers/service/UserService.java
package com.shihabmahamud.eshoppers.service;

import com.shihabmahamud.eshoppers.domain.User;
import com.shihabmahamud.eshoppers.dto.LoginDTO;
import com.shihabmahamud.eshoppers.dto.UserDTO;

public interface UserService {
    void saveUser(UserDTO userDTO);
    Boolean isNotUniqueUsername(UserDTO userDTO);
    User verifyUser(LoginDTO loginDTO);
}

File: eshoppers/src/main/java/com/shihabmahamud/eshoppers/service/UserService.kt
package com.shihabmahamud.eshoppers.service

import com.shihabmahamud.eshoppers.domain.User
import com.shihabmahamud.eshoppers.dto.LoginDTO
import com.shihabmahamud.eshoppers.dto.UserDTO

interface UserService {
    fun saveUser(userDTO: UserDTO)
    fun isNotUniqueUsername(userDTO: UserDTO): Boolean
    fun isNotUniqueEmail(userDTO: UserDTO): Boolean
    fun verifyUser(loginDTO: LoginDTO): User
}